tests/cpp/test_multidevice_communications.cpp

```diff
@@ -64,9 +67,13 @@ void CommunicationTest::validate(at::Tensor obtained, at::Tensor expected) {
 }
 
 TEST_P(CommunicationTest, Gather) {
-  IrContainer container;
-  auto communication = IrBuilder::createInContainer<Communication>(
-      &container, CommunicationType::Gather, full_mesh_, all_ranks_, kRoot);
+  hir::HostIrContainer container;
+  FusionGuard fg(&container);
+  auto* in = makeContigTensor(2);
+  in->setDeviceMesh(full_mesh_);
+  auto* out = ops::newValLike(in, in->dtype())->as<TensorView>();
+  auto communication = IrBuilder::create<Communication>(
```

Comment:
```suggestion
  auto* communication = IrBuilder::create<Communication>(
```
just to make the type more explicit.